pandas/core/groupby/generic.py

```diff
@@ -339,29 +341,99 @@ def _aggregate_multiple_funcs(self, arg, _level):
 
         return DataFrame(results, columns=columns)
 
-    def _wrap_series_output(self, output, index, names=None):
-        """ common agg/transform wrapping logic """
-        output = output[self._selection_name]
+    def _wrap_series_output(
+        self, output: Mapping[int, AnyArrayLike], index: Index, columns: Index
+    ) -> Union[Series, DataFrame]:
+        """
+        Wraps the output of a SeriesGroupBy operation into the expected result.
 
-        if names is not None:
-            return DataFrame(output, index=index, columns=names)
+        Parameters
+        ----------
+        output : dict[int, np.ndarray]
+            Dict where the key represents the columnar-index and the values are
+            the actual results. Must be ordered from 0..n
+        index : pd.Index
+            Index to apply to the output.
+        columns : pd.Index
+            Columns to apply to the output.
+
+        Returns
+        -------
+        Series or DataFrame
+
+        Notes
+        -----
+        In the vast majority of cases output and columns will only contain one
+        element. The exception is operations that expand dimensions, like ohlc.
+        """
+        assert len(output) == len(columns)
+        if len(output) > 1:
+            result = DataFrame(output, index=index)  # type: Union[Series, DataFrame]
```

Comment:
this seems like it shouldnt be necessary.  is a future version of mypy going to know how to handle this?

Reply:
Yea once we drop 35 we can change the syntax so that we define this before the if...else without declaring. For 35 compat you unfortunately have to annotate the first appearance, which is always a declaration